classes/Kohana/HTTP.php

```diff
@@ -180,14 +189,13 @@ public static function request_headers()
 
 		foreach ($_SERVER as $key => $value)
 		{
-			// If there is no HTTP header here, skip
-			if (strpos($key, 'HTTP_') !== 0)
+			// If there is HTTP header here
+			if (strpos($key, 'HTTP_') === 0)
 			{
-				continue;
+				// This is a dirty hack to ensure HTTP_X_FOO_BAR becomes x-foo-bar
+				$key = str_replace('_', '-', substr($key, 5));
+				$headers[$key] = $value;
```

Comment:
Why change this? It is recommended to fail early. So I would rather have the `continue;` back if possible.